### PR TITLE
Adding Mastodon to VTB

### PR DESCRIPTION
**Pull request by kyleeswanson**
Adding Mastodon to VTB per Guillaume's instruction

### CODE REVIEW

File: doc/content/seismic_analysis/mastoson.md
Comment (GiudGiud) [inline doc/content/seismic_analysis/mastoson.md:1]: mastoson -> npp_building.md

File: seismic_analysis/basemat_with_isolators_new.i [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+    block = 'upper_rigid_elems'
+    area = 130.06
+    Iy = 24166.729
+    Iz = 24166.729
+    y_orientation = '0.0 0.0 1.0'
+  []
+[]
+
+[Kernels]
+  [DynamicTensorMechanics]
```
[the remark this answers is not in the document]
Comment (kyleeswanson) [reply seismic_analysis/basemat_with_isolators_new.i:117]: What is the syntax? I'm not seeing any other Kernel types that are relevant

[the remark this answers is not in the document]
Comment (GiudGiud) [reply seismic_analysis/basemat_with_isolators_new.i:117]: see the dynamics input for examples
https://github.com/idaholab/moose/blob/next/modules/solid_mechanics/test/tests/dynamics/time_integration/newmark_action.i

File: seismic_analysis/mesh/full_structure_with_isolators_new.e
Comment (GiudGiud) [inline seismic_analysis/mesh/full_structure_with_isolators_new.e:1]: let's use LFS for that. I ll show you tomorrow

File: seismic_analysis/tests
Comment (GiudGiud) [inline seismic_analysis/tests:1]: tests all pass, good job

File: seismic_analysis/tests [hunk 2]
```diff
@@ -0,0 +1,59 @@
+[Tests]
+  [seismic_analysis_syntax1a]
+    type = RunApp
+    input = basemat_with_isolators_new.i
+    method = opt
```
Comment (GiudGiud) [inline seismic_analysis/tests:5]:
```suggestion
```
pls remove those from syntax tests, they are cheap

File: doc/content/seismic_analysis/npp_building/npp_building_pt2.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+!col-end!
+
+!row-end!
+
+!media media/seismic_analysis/rv_bot.png
+      style=width:50%
+      id=fig:rv_bot
+      caption=Head-supported reactor vessel as seen from the bottom.
+
+All the materials in the building are modeled using a linear elastic material and 3D 8-noded HEX elements, except for the FP isolators, which are modeled using two noded link elements with the FP isolator material model see [theory manual](https://mooseframework.inl.gov/mastodon/manuals/theory/index.html) and [user manual](https://mooseframework.inl.gov/mastodon/manuals/user/index.html) for more information. The reactor vessel is modeled as a thin cylindrical vessel using continuum elements. The fluid inside the reactor vessel is modeled using a linear elastic material with a very small shear modulus to reproduce fluid-like behavior. A more comprehensive, fluid-structure interaction (FSI) approach is also possible in MASTODON to model the fluid. More information on FSI modeling in MOOSE and MASTODON is provided [here](https://mooseframework.inl.gov/modules/fsi/index.html). Further information on the building model is also provided in [!citet](inl-ext-20-59608).
```
Comment (GiudGiud) [inline doc/content/seismic_analysis/npp_building/npp_building_pt2.md:37]:
```suggestion
All the materials in the building are modeled using a linear elastic material and 3D 8-noded HEX elements, except for the FP isolators, which are modeled using two noded link elements with the FP isolator material model see MASTODON [theory manual](https://mooseframework.inl.gov/mastodon/manuals/theory/index.html) and [user manual](https://mooseframework.inl.gov/mastodon/manuals/user/index.html) for more information. The reactor vessel is modeled as a thin cylindrical vessel using continuum elements. The fluid inside the reactor vessel is modeled using a linear elastic material with a very small shear modulus to reproduce fluid-like behavior. A more comprehensive, fluid-structure interaction (FSI) approach is also possible in MASTODON to model the fluid. More information on FSI modeling in MOOSE and MASTODON is provided [here](https://mooseframework.inl.gov/modules/fsi/index.html). Further information on the building model is also provided in [!citet](inl-ext-20-59608).
```

File: seismic_analysis/mesh/full_structure_with_isolators_new.e
Comment (GiudGiud) [inline seismic_analysis/mesh/full_structure_with_isolators_new.e:1]: this is too big let's use LFS for it

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Review by GiudGiud — COMMENTED** [review]
fails the vtb tests now. 
It looks like a file got lost in the folder change

**Comment by GiudGiud** [comment]
@cbolisetti if you have time to take a look

**Comment by cbolisetti** [comment]
@GiudGiud and @kyleeswanson looks good to me. Can you please also mention somewhere that this was adopted from the list of examples on the Mastodon website and link it to the examples page? Perhaps at the beginning of both Part 1 and Part 2?

**Comment by GiudGiud** [comment]
good idea. Let's also add Chandu as the POC on top of the relevant page, as well as a link to the models (from the doc page)

File: doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # Seismic analysis of a base-isolated nuclear power plant building
 
+This model was adopted from the list of examples on the [MASTODON](https://mooseframework.inl.gov/mastodon/examples/index.html) website.
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md:3]: add a link to the inputs on the virtual test bed
somewhere like https://github.com/idaholab/virtual_test_bed/tree/devel/msr/msr_generic/seismic_analysis

File: doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt2.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
+# Seismic analysis of a base-isolated nuclear power plant building
+
+This model was adopted from the list of examples on the [MASTODON](https://mooseframework.inl.gov/mastodon/examples/index.html) website.
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt2.md:3]: same

File: doc/content/msr/msr_generic/seismic_analysis/index.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+
+This model was adopted from the list of examples on the [MASTODON](https://mooseframework.inl.gov/mastodon/examples/index.html) website. For further information regarding [MASTODON](https://mooseframework.inl.gov/mastodon/index.html) please contact Chandu Bolisetti at chandrakanth.bolisetti@inl.gov.
+
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/index.md:6]: add a tag. see other model doco for the tagging syntax

**Comment by GiudGiud** [comment]
the tags are not displaying
https://mooseframework.inl.gov/virtualtestbed/docs/PRs/396/site/resources/filter/index.html

File: doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+!tag name=Seismic Analysis Model pairs=reactor_type:MSR
+                       reactor:generic_msr
+                       geometry:reactor_building
+                       codes_used:MASTODON
+                       computing_needs:HPC
+                       fiscal_year:2021
+                       sponsor:DOE Office of Technology Transitions (OTT)
+                       institution:INL
+                       simulation_type:seismic analysis
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md:13]:
```suggestion
```

File: doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt2.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+!tag name=Seismic Analysis Model pairs=reactor_type:MSR
+                       reactor:generic_msr
+                       geometry:reactor_building
+                       codes_used:MASTODON
+                       computing_needs:HPC
+                       fiscal_year:2021
+                       sponsor:DOE Office of Technology Transitions (OTT)
+                       institution:INL
+                       simulation_type:seismic analysis
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt2.md:13]:
```suggestion
```

File: doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md (hunk 7 above)
Comment (GiudGiud) [reply doc/content/msr/msr_generic/seismic_analysis/npp_building/npp_building_pt1.md:13]: no need to duplicate tags

File: doc/content/msr/msr_generic/seismic_analysis/index.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+This model was adopted from the list of examples on the [MASTODON](https://mooseframework.inl.gov/mastodon/examples/index.html) website. For further information regarding [MASTODON](https://mooseframework.inl.gov/mastodon/index.html) please contact Chandu Bolisetti at chandrakanth.bolisetti@inl.gov.
```
Comment (GiudGiud) [inline doc/content/msr/msr_generic/seismic_analysis/index.md:5]: Chandu is a short version of his name

use Dr. Bolisetti instead

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
The LFS include is still not working properly. Let s fix this tomorrow

**Comment by GiudGiud** [comment]
Test failures unrelated

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**